test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -176,3 +176,114 @@ def test_one_core_remove(cache_mode):
+        core_dev.create_partitions([Size(1, Unit.GibiByte)] * 2)
+        core_part1 = core_dev.partitions[0]
+        core_part2 = core_dev.partitions[1]
+        Udev.disable()
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Change sequential cutoff policy to 'never'."):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        if cache_mode == CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        elif cache_mode != CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input("/dev/urandom")
+                  .output(core1.system_path)
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        core1_occupied_blocks_before = core1.get_occupancy()
+
+    with TestRun.step("Check if the remaining core is able to use cache."):
+        try:
+            if cache_mode == CacheModeTrait.InsertRead:
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+            elif cache_mode != CacheModeTrait.InsertRead:
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
```

Comment:
You could leave preparing dd command inside `if` block, and execute rest of code unconditional.